Dreambooth: Ready to go!

d8ahazard opened the pull request:
Okay, this is the third submission of this.

Everything should work now, but may require some horsepower to do so. It can theoretically work on 10GB GPU, possibly 8 if the user sets up WSL2 and xformers and other stuff, but that will be left outside the scope of this project.

Point is, I've tested this quite thoroughly now, and everything does what it should do in a pretty efficient manner.

You can currently:

Fine-tune any existing checkpoint. Load it from the UI, configure settings, go.
Reload any existing set of training data from the UI, including prompts, directories, everything.
Train with or without "prior preservation loss".
Optionally train the text encoder as well, which promises better results with human subjects, same with PPL.
Auto-convert diffuser data back to checkpoint data.

Future things to implement (once initial is merged):
Multiple subjects at once.
Auto reload SD checkpoint list.
Add a "cooldown" option where you can pause after N steps to give your GPU a break, then resume again after N seconds/minutes.

Final submission, replaces #2002

d8ahazard committed:
Implement DreamBooth

It definitely works.

d8ahazard commented:
@AUTOMATIC1111 - Please give this a look and merge.

d8ahazard committed:
Update dreambooth.py

IdiotSandwichTheThird commented:
Have you considered adding the prompt template file for this? There are some reports that this increases quality, with forks like https://github.com/victorchall/EveryDream-trainer

LaNsHoR commented:
With this last version I get a javascript error when I click "Train", and nothing happens:

```
VM608:3 Uncaught (in promise) ReferenceError: start_training_dreambooth is not defined
    at eval (eval at <anonymous> (index.9828d028.js:54:2891), <anonymous>:3:14)
    at index.9828d028.js:56:4850
    at HTMLButtonElement.<anonymous> (index.9828d028.js:55:2100)
    at index.9828d028.js:4:1266
    at Array.forEach (<anonymous>)
    at HTMLButtonElement.Zn (index.9828d028.js:4:1253)
    at HTMLButtonElement._ (index.88652f98.js:1:2092)
    at index.9828d028.js:4:1266
    at Array.forEach (<anonymous>)
    at HTMLButtonElement.Zn (index.9828d028.js:4:1253)

d8ahazard committed:
DUH

Forgot to add the JS

0xItx commented:
Awesome job on this :)

Would you like me to rework d8ahazard/stable-diffusion-webui#2 into conversion.py?
From what limited testing I did, I got better results when using the model's own config, in case it was present. I would guess that using model's associated VAE may help too.

Evil-Dragon commented:
Clean install:
`Traceback (most recent call last):
  File "G:\stable-diffusion-webui-DreamBooth_V2\launch.py", line 228, in <module>
    start_webui()
  File "G:\stable-diffusion-webui-DreamBooth_V2\launch.py", line 222, in start_webui
    import webui
  File "G:\stable-diffusion-webui-DreamBooth_V2\webui.py", line 14, in <module>
    import modules.extras
  File "G:\stable-diffusion-webui-DreamBooth_V2\modules\extras.py", line 18, in <module>
    from modules.ui import plaintext_to_html
  File "G:\stable-diffusion-webui-DreamBooth_V2\modules\ui.py", line 24, in <module>
    from modules.dreambooth import dreambooth, conversion
  File "G:\stable-diffusion-webui-DreamBooth_V2\modules\dreambooth\dreambooth.py", line 15, in <module>
    from accelerate import Accelerator
ModuleNotFoundError: No module named 'accelerate'`

Evil-Dragon commented:
Sadly it still seems like it's still out of reach for 12GB VRAM users on Windows.

So close: RuntimeError: CUDA out of memory. Tried to allocate 50.00 MiB (GPU 0; 12.00 GiB total capacity; 11.10 GiB already allocated; 0 bytes free; 11.22 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

coltography commented:
is there any way to test this before it's been merged?

MartinCairnsSQL commented:
@coltography You can create a new folder and open it in command shell and use the one of the commands below to install it to that new folder.
`git clone https://github.com/d8ahazard/stable-diffusion-webui.git`
or if you have github cli installed
`gh repo clone d8ahazard/stable-diffusion-webui`

0xItx commented:
If you follow @MartinCairnsSQL, make sure to `git checkout DreamBooth_V2` afterwards to get the correct branch.
Alternatively `gh pr checkout 3995` (with the [GitHub CLI](https://cli.github.com/)) or follow https://stackoverflow.com/a/30584951

d8ahazard commented:
> Sadly it still seems like it's still out of reach for 12GB VRAM users on Windows.
> 
> So close: `RuntimeError: CUDA out of memory. Tried to allocate 50.00 MiB (GPU 0; 12.00 GiB total capacity; 11.10 GiB already allocated; 0 bytes free; 11.22 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF`
> 
> and CPU option basically spits out: `ValueError: AcceleratorState has already been initialized and cannot be changed, restart your runtime completely and pass `cpu=True`to`Accelerate()`.`
> 
> Can't play any more with it tonight, I have work early in the morning.

Not entirely? If you run under WSL*2* and properly configure deepspeed, 8bit-adam, and accelerate; then skip the "webui.sh" file and run with accelerate launch launch.py, it should be runnable on GPU for 12GB. Might need to disable training of the text encoder under "advanced" tho.

fredconex commented:
Unfortunately still can't use this, running on Win11/WSL with 12gb 3080ti

Starting Dreambooth training...
 VRAM cleared.
 Allocated: 0.0GB
 Reserved: 0.0GB

 Loaded model.
 Allocated: 0.0GB
 Reserved: 0.0GB

Exception importing 8bit adam: 'NoneType' object has no attribute 'cuDeviceGetCount'
 Scheduler Loaded
 Allocated: 0.0GB
 Reserved: 0.0GB

***** Running training *****
  Num examples = 12
  Num batches each epoch = 12
  Num Epochs = 84
  Instantaneous batch size per device = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Gradient Accumulation steps = 1
  Total optimization steps = 1000
  Total target lifetime optimization steps = 1000
 CPU: False Adam: False, Prec: fp16, Prior: False, Grad: True, TextTr: True
 Allocated: 3.8GB
 Reserved: 3.9GB

Steps:   0%|                                                                                   | 0/1000 [00:00<?, ?it/s]  First unet step completed.
 Allocated: 3.8GB
 Reserved: 3.9GB

 Caught exception.
 Allocated: 11.1GB
 Reserved: 11.2GB

Exception training db: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 12.00 GiB total capacity; 11.13 GiB already allocated; 0 bytes free; 11.23 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Traceback (most recent call last):
  File "/home/fred/stable-diffusion-webui/modules/dreambooth/dreambooth.py", line 491, in train
    optimizer.step()
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/accelerate/optimizer.py", line 134, in step
    self.scaler.step(self.optimizer, closure)
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/cuda/amp/grad_scaler.py", line 341, in step
    retval = self._maybe_opt_step(optimizer, optimizer_state, *args, **kwargs)
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/cuda/amp/grad_scaler.py", line 288, in _maybe_opt_step
    retval = optimizer.step(*args, **kwargs)
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/optim/lr_scheduler.py", line 68, in wrapper
    return wrapped(*args, **kwargs)
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/optim/optimizer.py", line 140, in wrapper
    out = func(*args, **kwargs)
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/optim/adamw.py", line 147, in step
    state['exp_avg'] = torch.zeros_like(p, memory_format=torch.preserve_format)
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 12.00 GiB total capacity; 11.13 GiB already allocated; 0 bytes free; 11.23 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

 CLEANUP:
 Allocated: 11.1GB
 Reserved: 11.2GB

 Cleanup Complete.
 Allocated: 11.0GB
 Reserved: 11.2GB

Steps:   0%|                                                                                   | 0/1000 [00:05<?, ?it/s] Training completed, reloading SD Model.
 Allocated: 0.0GB
 Reserved: 7.8GB

Memory output: {'VRAM cleared.': '0.0/0.0GB', 'Loaded model.': '0.0/0.0GB', 'Scheduler Loaded': '0.0/0.0GB', 'CPU: False Adam: False, Prec: fp16, Prior: False, Grad: True, TextTr: True ': '3.8/3.9GB', ' First unet step completed.': '3.8/3.9GB', 'Caught exception.': '11.1/11.2GB', 'CLEANUP: ': '11.1/11.2GB', 'Cleanup Complete.': '11.0/11.2GB', 'Training completed, reloading SD Model.': '0.0/7.8GB'}
Re-applying optimizations...
Applying cross attention optimization (Doggettx).
Returning result: Training finished. Total lifetime steps: 0

d8ahazard commented:
CPU Adam is erroring.

On Sun, Oct 30, 2022, 4:17 PM Alfredo Fernandes ***@***.***>
wrote:

> Unfortunately still can't use this, running on Win11/WSL with 12gb 3080ti
>
> Starting Dreambooth training...
> VRAM cleared.
> Allocated: 0.0GB
> Reserved: 0.0GB
>
> Loaded model.
> Allocated: 0.0GB
> Reserved: 0.0GB
>
> Exception importing 8bit adam: 'NoneType' object has no attribute
> 'cuDeviceGetCount'
> Scheduler Loaded
> Allocated: 0.0GB
> Reserved: 0.0GB
>
> ***** Running training *****
> Num examples = 12
> Num batches each epoch = 12
> Num Epochs = 84
> Instantaneous batch size per device = 1
> Total train batch size (w. parallel, distributed & accumulation) = 1
> Gradient Accumulation steps = 1
> Total optimization steps = 1000
> Total target lifetime optimization steps = 1000
> CPU: False Adam: False, Prec: fp16, Prior: False, Grad: True, TextTr: True
> Allocated: 3.8GB
> Reserved: 3.9GB
>
> Steps: 0%| | 0/1000 [00:00<?, ?it/s] First unet step completed.
> Allocated: 3.8GB
> Reserved: 3.9GB
>
> Caught exception.
> Allocated: 11.1GB
> Reserved: 11.2GB
>
> Exception training db: CUDA out of memory. Tried to allocate 20.00 MiB
> (GPU 0; 12.00 GiB total capacity; 11.13 GiB already allocated; 0 bytes
> free; 11.23 GiB reserved in total by PyTorch) If reserved memory is >>
> allocated memory try setting max_split_size_mb to avoid fragmentation. See
> documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> Traceback (most recent call last):
> File "/home/fred/stable-diffusion-webui/modules/dreambooth/dreambooth.py",
> line 491, in train
> optimizer.step()
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/accelerate/optimizer.py",
> line 134, in step
> self.scaler.step(self.optimizer, closure)
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/cuda/amp/grad_scaler.py",
> line 341, in step
> retval = self._maybe_opt_step(optimizer, optimizer_state, *args, **kwargs)
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/cuda/amp/grad_scaler.py",
> line 288, in _maybe_opt_step
> retval = optimizer.step(*args, **kwargs)
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/optim/lr_scheduler.py",
> line 68, in wrapper
> return wrapped(*args, **kwargs)
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/optim/optimizer.py",
> line 140, in wrapper
> out = func(*args, **kwargs)
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/autograd/grad_mode.py",
> line 27, in decorate_context
> return func(*args, **kwargs)
> File
> "/home/fred/anaconda3/envs/diffusers/lib/python3.9/site-packages/torch/optim/adamw.py",
> line 147, in step
> state['exp_avg'] = torch.zeros_like(p, memory_format=torch.preserve_format)
> torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 20.00
> MiB (GPU 0; 12.00 GiB total capacity; 11.13 GiB already allocated; 0 bytes
> free; 11.23 GiB reserved in total by PyTorch) If reserved memory is >>
> allocated memory try setting max_split_size_mb to avoid fragmentation. See
> documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
>
> CLEANUP:
> Allocated: 11.1GB
> Reserved: 11.2GB
>
> Cleanup Complete.
> Allocated: 11.0GB
> Reserved: 11.2GB
>
> Steps: 0%| | 0/1000 [00:05<?, ?it/s] Training completed, reloading SD
> Model.
> Allocated: 0.0GB
> Reserved: 7.8GB
>
> Memory output: {'VRAM cleared.': '0.0/0.0GB', 'Loaded model.':
> '0.0/0.0GB', 'Scheduler Loaded': '0.0/0.0GB', 'CPU: False Adam: False,
> Prec: fp16, Prior: False, Grad: True, TextTr: True ': '3.8/3.9GB', ' First
> unet step completed.': '3.8/3.9GB', 'Caught exception.': '11.1/11.2GB',
> 'CLEANUP: ': '11.1/11.2GB', 'Cleanup Complete.': '11.0/11.2GB', 'Training
> completed, reloading SD Model.': '0.0/7.8GB'}
> Re-applying optimizations...
> Applying cross attention optimization (Doggettx).
> Returning result: Training finished. Total lifetime steps: 0
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1296352993>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMO4NF36JMDREDIHG4TGATWF3QWTANCNFSM6AAAAAARSMEDCU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ArcticFaded commented:
fred you have xformers enabled right?

fredconex commented:
@d8ahazard any idea how to fix it? or where I should look at?

@ArcticFaded uh no xformers, I get bunch of errors when I enable the argument on webui, but I'm using shivam repo for dreambooth without issues that way

pravindahal commented:
Hi! I want to test this on Linux with RTX 4000 but I don't know what the Model dropdown is supposed to be. It is empty:
![image](https://user-images.githubusercontent.com/894731/198913310-4b75e4e7-fa24-4a50-959a-091dde2e7118.png)

What should I do so that there are models here?

Also, I found that there are missing dependencies in requirements.txt which makes the entire app fail to load if they are not installed. The following should be added to requirements.txt:
- facenet-pytorch
- accelerate

Lalimec commented:
> accelerate
create a model from "create model' tab 😃

webhead2oo9 commented:
It looks like the preprocess images area does not work for me, however everything else (so far) seems to work fine.

When trying any function in preprocess, this error is thrown.

```
Arguments: ('C:\\automaticdreamtest\\stable-diffusion-webui\\me', 'C:\\automaticdreamtest\\stable-diffusion-webui\\metest', True, True, False) {}
Traceback (most recent call last):
  File "C:\automaticdreamtest\stable-diffusion-webui\modules\ui.py", line 186, in f
    res = list(func(*args, **kwargs))
  File "C:\automaticdreamtest\stable-diffusion-webui\webui.py", line 53, in f
    res = func(*args, **kwargs)
  File "C:\automaticdreamtest\stable-diffusion-webui\modules\textual_inversion\ui.py", line 19, in preprocess
    modules.textual_inversion.preprocess.preprocess(*args)
TypeError: preprocess() missing 3 required positional arguments: 'process_flip', 'process_split', and 'process_caption'
```

Evil-Dragon commented:
bitsandbytes also needs to be installed manually but even when it's installed it throws "Exception importing 8bit adam: too many values to unpack (expected 5)"

As for WSL2, I haven't the faintest idea what i'm doing with that. I feel that the 10GB VRAM requirement that you stated isn't going to be possible without jumping through a lot of hoops which the average end user isn't going to know what to do. I mean CPU training works at least but there is still some work to get GPU training working without loads of messing around.

Sorry to be that 'guy' but i'm not as tech savvy as I once was.

Summersoff commented:
I think this is not ready for directly use yet. There are too many uncontrollable factors.

bmaltais commented:
You can follow my instructions on how to install xformers and Adam8bit support directly in windows...

https://github.com/bmaltais/kohya_ss

It is another dreambooth solution that work even with 8GB VRAM on windows as it does some special things to limit memory usage.

d8ahazard committed:
Add accelerate requirement

d8ahazard commented:
> I think this is not ready for directly use yet. There are too many uncontrollable factors. Unless bitsandbytes can locally support on windows, it will be possible to run DB locally without having to mess with WSL or something.

Aside from needing to update "Preprocess" due to the UI being updated in the 20+ days this has been an existing PR, what is not ready for use yet?

If you're running on a slow GPU, you need to install some extra requirements, which are out of scope of this project. If you're running on a decent GPU, it should just work.

Lalimec commented:
Training ends after 1 iteration with 3090, what might be the issue? And also how do i use pre-generated classification images, do i just state the path and number of images? Thank you for the great work btw.

futurevessel commented:
I can use Shivam's repo with 3060 12gb if I enable accelerate and 8-bit adam, I can even do 'train_text_encoder' if I enable gradient_checkpointing. I don't see this working ever in this WebUI though, since just launching it reserves ~2.8gb VRAM, which is VRAM that Dreambooth will need to work on a 12gb VRAM system, unless you enable deepspeed, which I've never tried (is there a steep performance penalty ?)

Perhaps you can have a separate launcher for Dreambooth which omits whatever is loaded into vram by default.

YakuzaSuske commented:
> I can use Shivam's repo with 3060 12gb if I enable accelerate and 8-bit adam, I can even do 'train_text_encoder' if I enable gradient_checkpointing. I don't see this working ever in this WebUI though, since just launching it reserves ~2.8gb VRAM, which is VRAM that Dreambooth will need to work on a 12gb VRAM system, unless you enable deepspeed, which I've never tried (is there a steep performance penalty ?)
> 
> Perhaps you can have a separate launcher for Dreambooth which omits whatever is loaded into vram by default.

Can post the link to the repo?

d8ahazard commented:
> I can use Shivam's repo with 3060 12gb if I enable accelerate and 8-bit adam, I can even do 'train_text_encoder' if I enable gradient_checkpointing. I don't see this working ever in this WebUI though, since just launching it reserves ~2.8gb VRAM, which is VRAM that Dreambooth will need to work on a 12gb VRAM system, unless you enable deepspeed, which I've never tried (is there a steep performance penalty ?)
> 
> Perhaps you can have a separate launcher for Dreambooth which omits whatever is loaded into vram by default.

Except for I clear the VRAM used by the webUI before starting training.

I changed the codebase from using Shivam's repo back to main because I didn't want to support requiring custom diffusers, but if this won't work because of that change, then I can probably switch it back. But it has been testing by @bmaltais on 12GB and proven working.

futurevessel commented:
Ah, that sounds fantastic, I would love using a 'all-in-one' solution like this repo for Dreambooth training since I do everything else with it. Was this running on 12gb without deepspeed ? Here's hoping this is merged soon! Thanks for your and bmailtais hard work on this.

ArcticFaded commented:
I dont feel like this needs to service everyone to be a part of the UI - being able to come in under 24GB is already good enough imo

webhead2oo9 commented:
My 4090 (24GB) can run this, I ran a few through it, even if my results were very hit and miss compared to the joepenna one.

0xdevalias commented:
Potentially Related (to add backlinks into all the seemingly relevant threads):

- https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/914
- https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/1429
- https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/1734
- https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/3586
- https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/2002
  - https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/2002#issuecomment-1296308600
    - > Closing, opening new PR to squash commits and make it clean.

Baquara commented:
Hello

Even after using accelerate I am still getting CUDA out of memory errors on my 12gb RTX 3060. Are there any tutorials or guides on how I should configure `accelerate config` on a 12gb card? Most guides I see are about configuring a parameters file, but `accelerate config` asks a bunch of questions.

If anyone with a limited VRAM card that can get this PR running can provide a step by step guide it would be appreciated.

AUTOMATIC1111 commented:
This should be an extension. Regardless of how much VRAM it uses or whether it works well, it's a huge chunk of code that I can't and don't want to study. It's your creation and you should have a repo with it. Extension support is coming along very well and users who want it will be able to use install it from UI.

ZeroCool22 commented:
> I dont feel like this needs to service everyone to be a part of the UI - being able to come in under 24GB is already good enough imo

Sure you say that because you have a good GPU, but this should aim to make it run for ppl with low VRAM too like https://github.com/smy20011/dreambooth-gui does.

Also, if the installation will need to be thought **WSL** on Windows, there must be a **good guide** on the page of this repo.

This isn't visible on the **Extensions** yet...

![Screenshot_2](https://user-images.githubusercontent.com/13344308/199345581-3a801433-03bd-4b77-8b9e-94c276a74a0c.jpg)

78Alpha commented:
> You can follow my instructions on how to install xformers and Adam8bit support directly in windows...
> 
> https://github.com/bmaltais/kohya_ss
> 
> It is another dreambooth solution that work even with 8GB VRAM on windows as it does some special things to limit memory usage.

I wouldn't quite say 8 GB. Very close though.

`RuntimeError: CUDA out of memory. Tried to allocate 10.00 MiB (GPU 0; 8.00 GiB total capacity; 7.16 GiB already allocated; 0 bytes free; 7.28 GiB reserved in total by PyTorch)`

devBin28 commented:
How can i add this pf on my web-ui Extension tab? I can't find it, Not ready? My idle 4090 24gb GPU ready to go! pls merge or help!

pravindahal commented:
> How can i add this PR on my web-ui Extension tab? I can't find it, Not ready? My idle 4090 24gb GPU ready to go! pls merge or help!

You can download zip (Code>Download ZIP) from the following link and run it as a separate install:
https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2

edbergc7 commented:
> > How can i add this PR on my web-ui Extension tab? I can't find it, Not ready? My idle 4090 24gb GPU ready to go! pls merge or help!
> 
> @devBin28 You can download zip (Code>Download ZIP) from the following link and run it as a separate install: https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2

do we just download and run it via the webui-user.bat ?  will dreambooth tab be avaliable ?

jnpatrick99 commented:
Wow, looked great! So I merged this PR with master and tried using Dreambooth. First created a model based on SD-1.5 then generated class images (also tried without them). Pretty much standard settings. There were no errors.

But the quality of images generated using it is much lower than dreambooth colab standalone notebook (from ShivamShrirao repo). I trained on a cat, then I typed something like "sks cat in a bucket" but rendering this prompt with trained model gave me just cats (no bucket) pretty much the same as input images (same background, colors etc.).

Are there any manuals or hints how to use this PR? 

Thank you very much!

ZeroCool22 commented:
> Wow, looked great! So I merged this PR with master and tried using Dreambooth. First created a model based on SD-1.5 then generated class images (also tried without them). Pretty much standard settings. There were no errors.
> 
> But the quality of images generated using it is much lower than dreambooth colab standalone notebook (from ShivamShrirao repo). I trained on a cat, then I typed something like "sks cat in a bucket" but rendering this prompt with trained model gave me just cats (no bucket) pretty much the same as input images (same background, colors etc.).
> 
> Are there any manuals or hints how to use this PR?
> 
> Thank you very much!

Stop using **SKS** please.

jnpatrick99 commented:
> > Wow, looked great! So I merged this PR with master and tried using Dreambooth. First created a model based on SD-1.5 then generated class images (also tried without them). Pretty much standard settings. There were no errors.
> > But the quality of images generated using it is much lower than dreambooth colab standalone notebook (from ShivamShrirao repo). I trained on a cat, then I typed something like "sks cat in a bucket" but rendering this prompt with trained model gave me just cats (no bucket) pretty much the same as input images (same background, colors etc.).
> > Are there any manuals or hints how to use this PR?
> > Thank you very much!
> 
> Stop using **SKS** please.

Is that the solution to better results?

ZeroCool22 commented:
> > > Wow, looked great! So I merged this PR with master and tried using Dreambooth. First created a model based on SD-1.5 then generated class images (also tried without them). Pretty much standard settings. There were no errors.
> > > But the quality of images generated using it is much lower than dreambooth colab standalone notebook (from ShivamShrirao repo). I trained on a cat, then I typed something like "sks cat in a bucket" but rendering this prompt with trained model gave me just cats (no bucket) pretty much the same as input images (same background, colors etc.).
> > > Are there any manuals or hints how to use this PR?
> > > Thank you very much!
> > 
> > 
> > Stop using **SKS** please.
> 
> Is that the solution to better results?

Use a **unique word** that DB can't associate with any previous trained on the base model.

https://www.reddit.com/r/StableDiffusion/comments/yju5ks/from_one_of_the_original_dreambooth_authors_stop/

demirklvc commented:
> This will require a rewrite as an extension. Not sure how much work this will be.
> […](#)
> On Wed, Nov 2, 2022, 9:33 a.m. devBin28 ***@***.***> wrote: How can i add this pf on my web-ui Extension tab? I can't find it, Not ready? My idle 4090 24gb GPU ready to go! pls merge or help! — Reply to this email directly, view it on GitHub <[#3995 (comment)](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1300413359)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABZA34WY73WECL2GNDLA5D3WGJUSHANCNFSM6AAAAAARSMEDCU> . You are receiving this because you were mentioned.Message ID: ***@***.***>

please

LaNsHoR commented:
I've been using these a few days and it is awesome, the results are mindblowing once you find the right parameters and the right way to train it.

I think this adds a huge value to the main web-ui project, if it needs to be an extension, I can try to help with that @d8ahazard @AUTOMATIC1111. Thank you both for the work and the effort you have put in these projects.

chakalakasp commented:
As features go, for those with compatible cards, this is an incredible resource.  And in the years going forward, compatible cards will be much more the norm.  People love SD, but there is real art in putting your own objects/people/styles into SD.

ZeroCool22 commented:
So, what DB Repo he is using for this, what's the **minimum required VRAM**?

pytesin commented:
Nice thing, tested it with my 3090 (24GB) and it works in windows 10. The V-RAM is near max used but it runs. Would love to see this as a Extension one day :)

TWIISTED-STUDIOS commented:
> 

Would also like to see this as an extension someday instead of it being bundled in default

TWIISTED-STUDIOS commented:
> > Nice thing, tested it with my 3090 (24GB) and it works in windows 10. The V-RAM is near max used but it runs. Would love to see this as a Extension one day :)
> 
> If you enable 8-bit and fp16 how much vram it use on windows?

What would you suggest the best way to use this is without having to go through the whole process of re-downloading the entire requirements in the bat file as they are an exact replica of the main branch?  (Windows OS)

stysmmaker commented:
I've made a quick conversion of this to an extension. All I did was adjust a few imports and set up the callback to add the tab to the UI. Worked for me when using the standard 1.4 model and disabling gradient checkpointing. I don't have interest in maintaining it so I'm just dropping it here for anyone else to pick up if they so choose to. https://i.mmaker.moe/stable-diffusion-webui-dreambooth.zip

demosch commented:
> I've made a quick conversion of this to an extension. All I did was adjust a few imports and set up the callback to add the tab to the UI. Worked for me when using the standard 1.4 model and disabling gradient checkpointing. I don't have interest in maintaining it so I'm just dropping it here for anyone else to pick up if they so choose to. https://i.mmaker.moe/stable-diffusion-webui-dreambooth.zip

just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not?

stysmmaker commented:
> just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not?
Yep, just like all other extensions. Gradient checkpointing is in the Advanced dropdown on the Dreambooth tab.

demosch commented:
cool! gonna try now.

4lt3r3go commented:
> > just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not?
> 
> Yep, just like all other extensions. Gradient checkpointing is in the Advanced dropdown on the Dreambooth tab.

installed but no dreambooth tabs popped in the ui here

pytesin commented:
> > Nice thing, tested it with my 3090 (24GB) and it works in windows 10. The V-RAM is near max used but it runs. Would love to see this as a Extension one day :)
> 
> If you enable 8-bit and fp16 how much vram it use on windows?

I left all the settings to standard

pytesin commented:
> > > Nice thing, tested it with my 3090 (24GB) and it works in windows 10. The V-RAM is near max used but it runs. Would love to see this as a Extension one day :)
> > 
> > 
> > If you enable 8-bit and fp16 how much vram it use on windows?
> 
> What would you suggest the best way to use this is without having to go through the whole process of re-downloading the entire requirements in the bat file as they are an exact replica of the main branch? (Windows OS)
> 
> do you have an install guide?

I did exactly that,a second installation to not mess up the main thing. I only added a symlink for sd1.5 and sd1.4 for training purpose and thats it. SO i dont need to duplicate the Checkpointfiles.

ZeroCool22 commented:
> > > just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not?
> > 
> > 
> > Yep, just like all other extensions. Gradient checkpointing is in the Advanced dropdown on the Dreambooth tab.
> 
> installed but no dreambooth tabs popped in the ui here

Same, no DB TAB.

![Screenshot_2](https://user-images.githubusercontent.com/13344308/200077323-b0b908a8-d41c-493d-b0fa-37734c021e23.jpg)

fredconex commented:
> > > Nice thing, tested it with my 3090 (24GB) and it works in windows 10. The V-RAM is near max used but it runs. Would love to see this as a Extension one day :)
> > 
> > 
> > If you enable 8-bit and fp16 how much vram it use on windows?
> 
> I left all the settings to standard

I was actually asking with those options enabled, I have a 3080ti and I'm running out of vram with it on windows, but I can run shivam dreambooth on WSL2 without problems, so I was wondering as you have 24gb with those options where it would sit on vram usage.

stysmmaker commented:
Needs to be in it's own folder (just like the other extensions).

fredconex commented:
> > > > just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not?
> > > 
> > > 
> > > Yep, just like all other extensions. Gradient checkpointing is in the Advanced dropdown on the Dreambooth tab.
> > 
> > 
> > installed but no dreambooth tabs popped in the ui here
> 
> Same, no DB TAB.
> 
> ![Screenshot_2](https://user-images.githubusercontent.com/13344308/200077323-b0b908a8-d41c-493d-b0fa-37734c021e23.jpg)

create a folder and drop those folders inside
![image](https://user-images.githubusercontent.com/109859677/200077546-07745e7d-2349-402d-b6de-ce10c188c432.png)
![image](https://user-images.githubusercontent.com/109859677/200077593-eb4c53fc-1f76-48f3-b1b7-f264dd7727f7.png)

d8ahazard commented:
I've been out of town this week, but will be home tomorrow. I'll review and
move this to my repo as an extension in the next day or two.

On Fri, Nov 4, 2022, 2:26 PM Emilio ***@***.***> wrote:

> just putting this into extension folder? also where to disable gradient
> checkpointing. is this on by default or not?
>
> Yep, just like all other extensions. Gradient checkpointing is in the
> Advanced dropdown on the Dreambooth tab.
>
> installed but no dreambooth tabs popped in the ui here
>
> Same, no DB TAB.
>
> [image: Screenshot_2]
> <https://user-images.githubusercontent.com/13344308/200077323-b0b908a8-d41c-493d-b0fa-37734c021e23.jpg>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1304274027>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMO4NHC44CCB3K6MUVGGCDWGV5P3ANCNFSM6AAAAAARSMEDCU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ZeroCool22 commented:
> > > > > just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not?
> > > > 
> > > > 
> > > > Yep, just like all other extensions. Gradient checkpointing is in the Advanced dropdown on the Dreambooth tab.
> > > 
> > > 
> > > installed but no dreambooth tabs popped in the ui here
> > 
> > 
> > Same, no DB TAB.
> > ![Screenshot_2](https://user-images.githubusercontent.com/13344308/200077323-b0b908a8-d41c-493d-b0fa-37734c021e23.jpg)
> 
> create a folder and drop those folders inside ![image](https://user-images.githubusercontent.com/109859677/200077546-07745e7d-2349-402d-b6de-ce10c188c432.png) ![image](https://user-images.githubusercontent.com/109859677/200077593-eb4c53fc-1f76-48f3-b1b7-f264dd7727f7.png)

![Uploading Screenshot_3.jpg…]()

pytesin commented:
> > > > Nice thing, tested it with my 3090 (24GB) and it works in windows 10. The V-RAM is near max used but it runs. Would love to see this as a Extension one day :)
> > > 
> > > 
> > > If you enable 8-bit and fp16 how much vram it use on windows?
> > 
> > 
> > I left all the settings to standard
> 
> I was actually asking with those options enabled, I have a 3080ti and I'm running out of vram with it on windows, but I can run shivam dreambooth on WSL2 without problems, so I was wondering as you have 24gb with those options where it would sit on vram usage.

23,6 GB, but i was running some Programs in Background which could eat 1 or 2 GB of it, so on my machine it spikes to 21-22, then tends to train with 23,6GB filled (minus ~1/2 GB)

stysmmaker commented:
Use a name other than `dreambooth`, that's going to cause the circular reference since there's a folder in there already named that.

ZeroCool22 commented:
> Use a name other than `dreambooth`, that's going to cause the circular reference since there's a folder in there already named that.

If i change the name, i get error...

![Screenshot_4](https://user-images.githubusercontent.com/13344308/200079070-9776a909-a0c2-4bb3-961c-c437292b5b56.jpg)

stysmmaker commented:
Move those `.py` files back into a folder named `dreambooth`, within that folder you screencapped. It's supposed to match the directory structure of the archive I posted.

ZeroCool22 commented:
![Screenshot_5](https://user-images.githubusercontent.com/13344308/200081280-2f4aab8e-bc49-4782-8a79-fc5e5272af70.jpg)

**Initialization text = Instance Prompt?**

**Classification text = Man, woman, etc..?**

We can delete the " * "?

ZeroCool22 commented:
![Screenshot_6](https://user-images.githubusercontent.com/13344308/200082291-d3ef496e-ff93-4b34-835d-82962cba93e7.jpg)

ZeroCool22 commented:
![Screenshot_8](https://user-images.githubusercontent.com/13344308/200082487-b3f63eed-a024-42fb-8da3-0588296a9b6a.jpg)

Some explanation on this?

ZeroCool22 commented:
![Screenshot_9](https://user-images.githubusercontent.com/13344308/200082644-6ad59983-73e7-4708-8533-f91ed4ec0380.jpg)

Errors and more errors...

iznanka commented:
> ![Скриншот_9](https://user-images.githubusercontent.com/13344308/200082644-6ad59983-73e7-4708-8533-f91ed4ec0380.jpg)
> 
> Ошибки и еще раз ошибки...

I get this error if I forget to select a model for training from the drop-down list

iznanka commented:
> На этой неделе меня не было в городе, но завтра я буду дома. Я рассмотрю и перенесу это в свой репозиторий в качестве расширения в ближайшие день или два.
> […](#)
> On Fri, Nov 4, 2022, 2:26 PM Emilio ***@***.***> wrote: just putting this into extension folder? also where to disable gradient checkpointing. is this on by default or not? Yep, just like all other extensions. Gradient checkpointing is in the Advanced dropdown on the Dreambooth tab. installed but no dreambooth tabs popped in the ui here Same, no DB TAB. [image: Screenshot_2] <https://user-images.githubusercontent.com/13344308/200077323-b0b908a8-d41c-493d-b0fa-37734c021e23.jpg> — Reply to this email directly, view it on GitHub <[#3995 (comment)](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1304274027)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAMO4NHC44CCB3K6MUVGGCDWGV5P3ANCNFSM6AAAAAARSMEDCU> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Are looking forward to! I was already upset thinking that you abandoned the project. Thank you very much for your work. In my opinion you are doing the most important extra feature for SD-webui

ZeroCool22 commented:
```
Error loading script: main.py
Traceback (most recent call last):
  File "C:\Users\ZeroCool22\Desktop\Auto\modules\scripts.py", line 171, in load_scripts
    exec(compiled, module.__dict__)
  File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\scripts\main.py", line 2, in <module>
    from dreambooth import conversion, dreambooth
  File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\dreambooth\conversion.py", line 26, in <module>
    from dreambooth.dreambooth import get_db_models
  File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\dreambooth\dreambooth.py", line 15, in <module>
    from accelerate import Accelerator
ModuleNotFoundError: No module named 'accelerate'
```

iznanka commented:
> ```
> Error loading script: main.py
> Traceback (most recent call last):
>   File "C:\Users\ZeroCool22\Desktop\Auto\modules\scripts.py", line 171, in load_scripts
>     exec(compiled, module.__dict__)
>   File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\scripts\main.py", line 2, in <module>
>     from dreambooth import conversion, dreambooth
>   File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\dreambooth\conversion.py", line 26, in <module>
>     from dreambooth.dreambooth import get_db_models
>   File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\dreambooth\dreambooth.py", line 15, in <module>
>     from accelerate import Accelerator
> ModuleNotFoundError: No module named 'accelerate'
> ```
try installing dependencies

pip install ninja bitsandbytes
pip install facenet-pytorch
COMMANDLINE_ARGS="--exit" REQS_FILE="requirements.txt" python launch.py

4lt3r3go commented:
![image](https://user-images.githubusercontent.com/44883288/200142561-75c8f7a4-2b7f-4143-98e4-45ec91222663.png)

same error here
sorry i have no idea where to type those pip stuff
i tryed in anaconda, not sure if i did right
maybe i have to do cd d:\stablediffusion folder?
i got this
![image](https://user-images.githubusercontent.com/44883288/200142662-8a421401-53ed-4677-b831-144cae0b34fb.png)
then this 
![image](https://user-images.githubusercontent.com/44883288/200142680-95d9fb7d-c775-41ed-bcff-01ef20393770.png)
but still not popping in the UI, with that error i mentioned initially

ZeroCool22 commented:
> > ```
> > Error loading script: main.py
> > Traceback (most recent call last):
> >   File "C:\Users\ZeroCool22\Desktop\Auto\modules\scripts.py", line 171, in load_scripts
> >     exec(compiled, module.__dict__)
> >   File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\scripts\main.py", line 2, in <module>
> >     from dreambooth import conversion, dreambooth
> >   File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\dreambooth\conversion.py", line 26, in <module>
> >     from dreambooth.dreambooth import get_db_models
> >   File "C:\Users\ZeroCool22\Desktop\Auto\extensions\dreambooth\dreambooth\dreambooth.py", line 15, in <module>
> >     from accelerate import Accelerator
> > ModuleNotFoundError: No module named 'accelerate'
> > ```
> 
> try installing dependencies
> 
> pip install ninja bitsandbytes pip install facenet-pytorch COMMANDLINE_ARGS="--exit" REQS_FILE="requirements.txt" python launch.py
> 
> I didn’t manage to run deepspeed in windows, if you don’t have a 24 GB GPU (usually it takes 15.3-19), then for now it remains only to wait for optimizations. Or turn your attention to other projects designed for 12 GB GPU for training

Thx, you save me wasting time then, yeah i'm using this (with a 1080 TI): 

https://github.com/ShivamShrirao/diffusers/tree/main/examples/dreambooth

iznanka commented:
> ![изображение](https://user-images.githubusercontent.com/44883288/200142561-75c8f7a4-2b7f-4143-98e4-45ec91222663.png)
> 
> та же ошибка, извините, я понятия не имею, где набирать те пипы, которые я пробовал в анаконде, не уверен, правильно ли я сделал, может быть, мне нужно сделать папку cd d:\stablediffusion? я получил это:
> 
> ![изображение](https://user-images.githubusercontent.com/44883288/200142662-8a421401-53ed-4677-b831-144cae0b34fb.png)
> 
> тогда это:
> 
> ![изображение](https://user-images.githubusercontent.com/44883288/200142680-95d9fb7d-c775-41ed-bcff-01ef20393770.png)
> 
> но все еще не появляется в пользовательском интерфейсе, и я получил ту ошибку, о которой упоминал изначально

look in your requirements.txt file, accelerate should be the first one. try installing it separately

4lt3r3go commented:
> look in your requirements.txt file, accelerate should be the first one. try installing it separately

theres no accelerate in that txt
and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao
i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.

TWIISTED-STUDIOS commented:
> > look in your requirements.txt file, accelerate should be the first one. try installing it separately
> 
> theres no accelerate in that txt and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.

are you using the pull request or the extension that was posted here, but to activate the SD environment, go to your stable dif's main directory open terminal/cmd and run \venv\Scripts\activate.bat - I believe it is, then when you run pip install accelerate it will install into the environment that SD pulls from which is the same environment that the extension uses.

remusn3t commented:
> > look in your requirements.txt file, accelerate should be the first one. try installing it separately
> 
> theres no accelerate in that txt and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.

i know it's a stupid modification, but it works, i'm new to this stuff. i modified main.py from dreambooth, it automatically installs accelerate, for me it works with just that


[main.zip](https://github.com/AUTOMATIC1111/stable-diffusion-webui/files/9946478/main.zip)

iznanka commented:
> > look in your requirements.txt file, accelerate should be the first one. try installing it separately
> 
> theres no accelerate in that txt and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.

https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2
Download as zip, unpack and start. Work if you gpu 20GB+ or CPU train

4lt3r3go commented:
> > > look in your requirements.txt file, accelerate should be the first one. try installing it separately
> > 
> > 
> > theres no accelerate in that txt and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.
> 
> https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2 Download as zip, unpack and start. Work if you gpu 20GB+ or CPU train

Ok lets try his 
nope, nothing is working. I just give up 
errors:

![image](https://user-images.githubusercontent.com/44883288/200191145-b84800f5-f5cf-4f37-a0f0-e71468e7770b.png)


i tryed every single dreambooth offline methods around
the only working for me (but missing some features) is nmkd app.

TWIISTED-STUDIOS commented:
> > > > look in your requirements.txt file, accelerate should be the first one. try installing it separately
> > > 
> > > 
> > > theres no accelerate in that txt and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.
> > 
> > 
> > https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2 Download as zip, unpack and start. Work if you gpu 20GB+ or CPU train
> 
> Ok lets try his nope, nothing is working. I just give up. my 3090 can wait. errors:
> 
> ![image](https://user-images.githubusercontent.com/44883288/200191145-b84800f5-f5cf-4f37-a0f0-e71468e7770b.png)
> 
> i tryed every single dreambooth offline methods around the only working for me (but missing some features) is nmkd app.

Well that error is a simple fix to install torch, add --default-timeout=150 before the --extra-index-url inside the launch .py.

re launch and wait it will look like its frozen do not touch it in the background its downloading torch it just takes a while.

iznanka commented:
> 

no space on device?! :)))
tried https://github.com/smy20011/dreambooth-gui ?

4lt3r3go commented:
> 

yeah i realized that and redid from beginning on another drive
now having this error: 

![image](https://user-images.githubusercontent.com/44883288/200192659-1fb5ec24-bc50-4556-abc3-6d3b5d4ddb0c.png)

4lt3r3go commented:
> > 
> 
> no space on device?! :))) tried https://github.com/smy20011/dreambooth-gui ?

yeah i realized that and redid from beginning on another drive
now having this error:

![image](https://user-images.githubusercontent.com/44883288/200192692-5b28c025-c256-4bca-af5d-16be6f59dfdf.png)

0xdevalias commented:
How did you move things to another drive? That looks like you didn't clone the repo using git, or that you didn't move the hidden .git folder along with the rest of it if you copied the files somewhere.

chakalakasp commented:
I’ve done some testing and unfortunately the nmkd app output just sucks compared to other methods. You really do need prior loss preservation to get good results on people.

-Ryan McGinnis
***@***.***
[https://bigstormpicture.com](http://bigstormpicture.com)
GPG: 5C73 8727 EE58 786A 777C 4F1D B5AA 3FA3 486E D7AD

On Sun, Nov 6, 2022 at 13:34, 4lt3r3go ***@***.***> wrote:

>>>> look in your requirements.txt file, accelerate should be the first one. try installing it separately
>>>
>>> theres no accelerate in that txt and i have no idea how to install it properly. I have zero knowledge of python/pythorch/pystuff lmao i guess i'll just sit here with popcorn and wait you nerd guys come out with something easy to install and use for newbies like me.
>>
>> https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2 Download as zip, unpack and start. Work if you gpu 20GB+ or CPU train
>
> Ok lets try his
> nope, nothing is working. I just give up
> errors:
>
> [image](https://user-images.githubusercontent.com/44883288/200191145-b84800f5-f5cf-4f37-a0f0-e71468e7770b.png)
>
> i tryed every single dreambooth offline methods around
> the only working for me (but missing some features) is nmkd app.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1304875969), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AMNT5O4RYSST5IP7AFSWJ33WHAB33ANCNFSM6AAAAAARSMEDCU).
> You are receiving this because you commented.Message ID: ***@***.***>

4lt3r3go commented:
> I’ve done some testing and unfortunately the nmkd app output just sucks compared to other methods. You really do need prior loss preservation to get good results on people.
> […](#)

prior loss preservation? i dont see this thing in this repo here

```
(this discussion went too far and too many stuff/repos mentioned
to clarify to who is reading this for the first time, i tryed this repo and now i have dreambooth in the ui
https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2)
```

so I finally managed to make it work, doing from beginning again .. now i need to understand how to use it 🥲
this is way more complex than NMKD
![image](https://user-images.githubusercontent.com/44883288/200198082-74d74e38-fec3-424c-b957-bc8b61877b42.png)

d8ahazard commented:
Closing this, as I've now started a repo with a standalone extension based on ShivShiram's repo here:

https://github.com/d8ahazard/sd_dreambooth_extension

Please feel free to test and yell at me there. I've added requirements installer, multiple concept training via JSON, and moved some bit about.

UI still needs fixing, some stuff broken there, but it *should* be able to train a model for now.

ZeroCool22 commented:
> Closing this, as I've now started a repo with a standalone extension based on ShivShiram's repo here:
> 

So, we don't need anymore a 24gb of VRAM then?

d8ahazard commented:
> > Closing this, as I've now started a repo with a standalone extension based on ShivShiram's repo here:
> 
> So, we don't need anymore a 24gb of VRAM then?

Theoretically, no?

chakalakasp commented:
Prior preservation loss is something where you add a large number of images in the same class as the thing you were trying to train. So for training a person, you would generate a couple thousand images in the person class, or use one of the existing data sets that people have already generated. the dream booth training will then reference those images to try to contain the changes made to the model to the class of things you are training against and not bleed over much into other things. There are some other really handy tweaks that can be done as well, such as learning rate and doing text training, that seems to have a big impact. Long story short, using the vanilla plain Jane text-based encoder, you can achieve really good results. But most of the GUI tools that I have used or the google collab or the runpod scripts I have seen kinda suck in comparison

-Ryan McGinnis
***@***.***
[https://bigstormpicture.com](http://bigstormpicture.com)
GPG: 5C73 8727 EE58 786A 777C 4F1D B5AA 3FA3 486E D7AD

On Sun, Nov 6, 2022 at 16:22, 4lt3r3go ***@***.***> wrote:

>> I’ve done some testing and unfortunately the nmkd app output just sucks compared to other methods. You really do need prior loss preservation to get good results on people.
>> […](#)
>
> prior loss preservation? i dont see this thing in this repo here
>
> (this discussion went too far and too many stuff/repos mentioned
> to clarify to who is reading this for the first time, i tryed this repo and now i have dreambooth in the ui
> https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2)
>
> so I finally managed to make it work, doing from beginning again .. now i need to understand how to use it 🥲
> this is way more complex than NMKD
> [image](https://user-images.githubusercontent.com/44883288/200198082-74d74e38-fec3-424c-b957-bc8b61877b42.png)
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1304909515), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AMNT5O6SPCYSAYQHVZOZW4LWHAVQXANCNFSM6AAAAAARSMEDCU).
> You are receiving this because you commented.Message ID: ***@***.***>

ZeroCool22 commented:
> > > Closing this, as I've now started a repo with a standalone extension based on ShivShiram's repo here:
> > 
> > 
> > So, we don't need anymore a 24gb of VRAM then?
> 
> Theoretically, no?

Should be... https://github.com/ShivamShrirao/diffusers/tree/main/examples/dreambooth

4lt3r3go commented:
> Closing this, as I've now started a repo with a standalone extension based on ShivShiram's repo here:
> 
> https://github.com/d8ahazard/sd_dreambooth_extension
> 
> Please feel free to test and yell at me there. I've added requirements installer, multiple concept training via JSON, and moved some bit about.
> 
> UI still needs fixing, some stuff broken there, but it _should_ be able to train a model for now.

so this    https://github.com/d8ahazard/sd_dreambooth_extension
and this  https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2
are the same thing right?
one is extention only, other is merged auto1111 with the extention already installed?





> Prior preservation loss is something where you add a large number of images in the same class as the thing you were trying to train. So for training a person, you would generate a couple thousand images in the person class, or use one of the existing data sets that people have already generated. the dream booth training will then reference those images to try to contain the changes made to the model to the class of things you are training against and not bleed over much into other things. There are some other really handy tweaks that can be done as well, such as learning rate and doing text training, that seems to have a big impact. Long story short, using the vanilla plain Jane text-based encoder, you can achieve really good results. But most of the GUI tools that I have used or the google collab or the runpod scripts I have seen kinda suck in comparison
> […](#)
> -Ryan McGinnis ***@***.*** [[https://bigstormpicture.com](http://bigstormpicture.com)](https://bigstormpicture.com%5D(http://bigstormpicture.com)) GPG: 5C73 8727 EE58 786A 777C 4F1D B5AA 3FA3 486E D7AD
> On Sun, Nov 6, 2022 at 16:22, 4lt3r3go ***@***.***> wrote: > I’ve done some testing and unfortunately the nmkd app output just sucks compared to other methods. You really do need prior loss preservation to get good results on people. > […](#) prior loss preservation? i dont see this thing in this repo here (this discussion went too far and too many stuff/repos mentioned to clarify to who is reading this for the first time, i tryed this repo and now i have dreambooth in the ui https://github.com/d8ahazard/stable-diffusion-webui/tree/DreamBooth_V2) so I finally managed to make it work, doing from beginning again .. now i need to understand how to use it 🥲 this is way more complex than NMKD [image](https://user-images.githubusercontent.com/44883288/200198082-74d74e38-fec3-424c-b957-bc8b61877b42.png) — Reply to this email directly, [view it on GitHub]([#3995 (comment)](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1304909515)), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AMNT5O6SPCYSAYQHVZOZW4LWHAVQXANCNFSM6AAAAAARSMEDCU). You are receiving this because you commented.Message ID: ***@***.***>

thank you! i finally got the concept now
i think is just a matter of time someone will finally comes out with a detailed guide on how to operate properly
with all this settings now.. 
so, according to your words, as i understood well, the other method i was using (NMKD) produce a model that tend to "bleed the intended subject over into other things and class"
..in fact i noticed that every person i generate using those ckpt i made were kinda looking like the images i used for the training...

leoftm commented:
Thanks, dreambooth helped me learn.
There were a few errors that bothered me.
AttributeError: 'DreamboothConfig' object has no attribute 'seed'
I got this when the preview image was generated, so it may be because of the -1 seed in txt2img.
To skip preview generation
Generate a preview image every N steps, 0 to disable
to 0, I got a ZeroDivisionError.
I was able to finish the training to the end by setting it to a suitable unreachable value such as 1000000.

ZeroCool22 commented:
`Error no kernel image is available for execution on the device at line 167 in file D:\ai\tool\bitsandbytes\csrc\ops.cu`

![Screenshot_11](https://user-images.githubusercontent.com/13344308/200720905-c026cd27-a1bd-42f1-9176-1a12773eb8ad.jpg)
